### PR TITLE
Wednesday forms

### PR DESCRIPTION
**Pull request by ALynnThomas**
Added a form for user submission of a new store. Updates total row after user submission.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -3,88 +3,117 @@
...
-var alki = {
-  name: 'alki',
-  minCustomers:2,
-  maxCustomers:16,
-  avgCookies:4.6,
-  randomCustomers: function(){
-    return Math.floor(Math.random() * (16 - 2)) + 2;
-  },
-  cookiesPerHour: []
+function CookieStore(name, minCustomers, maxCustomers, avgCookies){
```
Comment (fraziermork) [inline app.js:6]: Nice!

File: app.js [hunk 2]
```diff
@@ -3,88 +3,117 @@
...
-function putSalesProjections(location){
-  console.log(location.name + '-projections');
-  var ulElement = document.getElementById(location.name + '-projections');
-
-  for(var ii = 0; ii < 15; ii++){
-    var liElement = document.createElement('li');
-    liElement.setAttribute('class', 'avgCookiesPerHour');
-    liElement.innerHTML = storeHours[ii] + ': ' + location.cookiesPerHour[ii] + ' cookies';
-    ulElement.appendChild(liElement);
+CookieStore.prototype.makeTheTable = function(tableToDrawInto){
```
Comment (fraziermork) [inline app.js:32]: Technically, this only draws a single row of the table.

File: app.js [hunk 3]
```diff
@@ -3,88 +3,117 @@
...
+    var hourlyCookieSales = document.createElement('td');
+    console.log(hourlyCookieSales);
+    hourlyCookieSales.textContent = this.cookiesPerHour[hours];
+    console.log(hourlyCookieSales);
+    hourlySalesRowEl.appendChild(hourlyCookieSales);
+  }
+  tableToDrawInto.appendChild(hourlySalesRowEl);
+};
+
+var tableEl = document.createElement('table');
```
Comment (fraziermork) [inline app.js:50]: It'll be helpful for you if you group global variables like this up at the very top.

File: app.js [hunk 4]
```diff
@@ -3,88 +3,117 @@
...
+var hoursHeaderEl = document.createElement('th');
+hoursRowEl.appendChild(hoursHeaderEl);
+for(var tableHours = 0; tableHours < storeHours.length; tableHours++){
+  var hoursDataEl = document.createElement('td');
+  hoursDataEl.textContent = storeHours[tableHours];
+  hoursRowEl.appendChild(hoursDataEl);
+}
+tableEl.appendChild(hoursRowEl);
+
+var stores = [firstAndPike, seatacAirport, seattleCenter, capitolHill, alki];
```
Comment (fraziermork) [inline app.js:65]: I would group this line up with the 5 lines where you instantiate these cookieStores.

File: app.js [hunk 5]
```diff
@@ -3,88 +3,117 @@
...
+  }
+  tableToDrawInto.appendChild(hourlySalesRowEl);
+};
+
+var tableEl = document.createElement('table');
+console.log(tableEl);
+var sectionEl = document.getElementById('cookieSalesTable');
+sectionEl.appendChild(tableEl);
+
+var hoursRowEl = document.createElement('tr');
```
Comment (fraziermork) [inline app.js:55]: This is the kind of stuff that would be good to break out into a function.

File: app.js [hunk 6]
```diff
@@ -3,88 +3,117 @@
...
+var stores = [firstAndPike, seatacAirport, seattleCenter, capitolHill, alki];
+
+console.log('==============EVENT LISTENERS===============');
+
+var newUserStores = [];
+var storeFormEl = document.getElementById('new-store-form');
+
+storeFormEl.addEventListener('submit', handleSubmit);
+
+function handleSubmit(event){
```
Comment (fraziermork) [inline app.js:74]: Great!

File: app.js [hunk 7]
```diff
@@ -3,88 +3,117 @@
...
+  hoursDataEl.textContent = storeHours[tableHours];
+  hoursRowEl.appendChild(hoursDataEl);
+}
+tableEl.appendChild(hoursRowEl);
+
+var stores = [firstAndPike, seatacAirport, seattleCenter, capitolHill, alki];
+
+console.log('==============EVENT LISTENERS===============');
+
+var newUserStores = [];
```
Comment (fraziermork) [inline app.js:69]: Put these globals up at the top too!

File: app.js [hunk 8]
```diff
@@ -3,88 +3,117 @@
...
+
+  var newStore = new CookieStore(name, minCustomers, maxCustomers, avgCookies);
+  stores.push(newStore);
+  getCookiesPerHour(newStore);
+  tableEl.deleteRow(-1);
+  newStore.makeTheTable(tableEl);
+  makeTotalRow();
+}
+
+for(var storeNumber = 0; storeNumber < stores.length; storeNumber++){
```
Comment (fraziermork) [inline app.js:96]: Break this out into a function! That way you can reuse it.

File: app.js [hunk 9]
```diff
@@ -3,88 +3,117 @@
...
+  getCookiesPerHour(stores[storeNumber]);
+  stores[storeNumber].makeTheTable(tableEl);
+}
+
+function makeTotalRow(){
```
Comment (fraziermork) [inline app.js:101]: Great!

File: app.js [hunk 10]
```diff
@@ -3,88 +3,117 @@
...
+  var totalRowEl = document.createElement('tr');
+  tableEl.appendChild(totalRowEl);
+  var totalHeaderEl = document.createElement('th');
+  totalHeaderEl.textContent = 'Totals';
+  totalRowEl.appendChild(totalHeaderEl);
+  var totalCookies = 0;
```
Comment (fraziermork) [inline app.js:107]: Can you think of a cleaner way to handle the totalCookies variable?

File: app.js [hunk 11]
```diff
@@ -3,88 +3,117 @@
...
   }
 }
 
-//call the functions to make the lists
-putSalesProjections(firstAndPike);
-putSalesProjections(seatacAirport);
-putSalesProjections(seattleCenter);
-putSalesProjections(capitolHill);
-putSalesProjections(alki);
+makeTotalRow();
```
Comment (fraziermork) [inline app.js:119]: If you break out the assorted pieces of functionality distributed throughout the file into functions, then you can group all the function calls down at the bottom--it'll be super readable!

File: appTwo.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
+console.log('==============EVENT LISTENERS===============');
```
Comment (fraziermork) [inline appTwo.js:1]: What is this file for?

File: index.html [hunk 13]
```diff
@@ -26,7 +26,26 @@
       <li>Capitol Hill</li>
       <li>Alki</li>
     </ul>
-    <p id="inspiration">Remember! Life might get hard, but always swim upstream!</p>
-    <script src="app.js" type="text/javascript"></script>
+    <div class="form">
```
Comment (fraziermork) [inline index.html:29]: I would avoid naming classes after element tags, especially if you're adding that class onto a different kind of element.

File: index.html [hunk 14]
```diff
@@ -26,7 +26,26 @@
...
+            <label for="avgCookies">Average Cookies/Customer</label>
+            <input type="text" min="0" step="any" name="avgCookies" value="">
+            <button type="submit" name="submitStoreInfo">Submit</button>
+        </fieldset>
+      </form>
+    </div>
+    <section id="cookieSalesTable">
+
+    </section>
+    <p id="inspiration">Remember! Life might get hard, but always swim upstream! Get it? Cause we serve salmon!</p>
```
Comment (fraziermork) [inline index.html:48]: Can you think of a cleaner way to format this p tag?

File: index.html [hunk 15]
```diff
@@ -26,7 +26,26 @@
       <li>Capitol Hill</li>
       <li>Alki</li>
     </ul>
-    <p id="inspiration">Remember! Life might get hard, but always swim upstream!</p>
-    <script src="app.js" type="text/javascript"></script>
+    <div class="form">
+      <form id="new-store-form">
+        <fieldset>
+          <legend>Enter New Cookie Store Info</legend>
+            <label for="newStoreName">Cookie Store Name:</label>
```
Comment (fraziermork) [inline index.html:33]: Technically, the for attribute needs to take the id of the form element it labels, not the name of it.

File: style.css [hunk 16]
```diff
@@ -25,13 +25,37 @@ h1 {
...
+  margin-left:auto;
+  margin-right: auto;
+  margin-top: 100px;
+}
+
+label, input{
+  display: block;
+}
+
+#new-store-form{
```
Comment (fraziermork) [inline style.css:53]: Try to style things with classes instead of ids.

**Review by fraziermork — COMMENTED** [review]
Nice work!

**Comment by fraziermork** [comment]
You have a lot of files in your project, and it's not entirely clear which one is which, or why they are there.  The purpose of version management is that you don't need to have files like appTwo--even if you delete everything in app and replace it with new stuff, you can find the old stuff in your git history.

File: index.html (hunk 15 above)
Comment (ALynnThomas) [reply index.html:33]: I'm really confused about what you mean by this. Could you help me with that tomorrow morning?

**Pull request merged**